include/kllvm/ast/AST.h

```diff
@@ -188,7 +178,7 @@ class KOREObjectSymbol : public KORESymbol {
   const std::vector<KOREObjectSort *> &getArguments() const {
     return arguments;
   }
-  const KOREObjectSort *getSort() const { return sort; }
```

Comment:
instead of removing the `const` version, I would suggest to overload with a non-`const` one. So keep
`const KOREObjectSort *getSort() const { return sort; }`
and add the overload
`KOREObjectSort *getSort() const { return sort; }`

Reply:
Pretty sure you can't have two overloads with the same signature and different return types, but it doesn't matter because KOREObjectSort * is implicitly convertible to const KOREObjectSort *.

Reply:
Oh yes, sorry I meant to suggest these as the overloaded methods:
`const KOREObjectSort *getSort() const { return sort; }`
`KOREObjectSort *getSort() { return sort; }`

I am positive that you can have overloads like above. This way if one has a `const KOREObjectSort *` or `const KOREObjectSort &`, they cannot get access to a non-const pointer to the member.

Reply:
Yes, you can overload on the constness of `this`, but not on return type. Since this would overload on this, you can also have a differing return type.